Updated description of cut and qcut

Ceay1 opened the pull request:
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] A colleague of mine complained about the quality of the `cut` description. I took a chance to update the description of `cut` and `qcut`. If there are some suggestions to improve the description - please let me know.

Ceay1 committed:
Update tile.py

pep8speaks commented:
Hello @Ceay1! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`pandas/core/reshape/tile.py`](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py):

> [Line 44:76](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L44): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 46:1](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L46): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 47:80](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L47): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 49:76](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L49): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 50:76](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L50): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 299:76](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L299): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 301:1](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L301): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 302:60](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L302): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 303:78](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L303): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 304:81](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L304): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 305:80](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L305): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace
> [Line 306:33](https://github.com/pandas-dev/pandas/blob/4bb31271110ed792c599977089cbc14eb0395176/pandas/core/reshape/tile.py#L306): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

attack68 commented on pandas/core/reshape/tile.py:
```diff
@@ -41,13 +41,15 @@ def cut(
-    age ranges. Supports binning into an equal number of bins, or a
-    pre-specified array of bins.
+    Transforms a variable with a large number of possibly continuous values 
+    into a smaller number of equal-width or predefined intervals, or bins.
```
this generates checks errors cause it doesn't conform to grammar style. also these first line snippets are used as summaries so should be short.

consider: `Label values into pre-defined or equal-width intervals, or bins`

Ceay1 replied:
Updated, shortened.

attack68 commented on pandas/core/reshape/tile.py:
```diff
@@ -41,13 +41,15 @@ def cut(
+    into a smaller number of equal-width or predefined intervals, or bins.
+    
+    Use `cut` when you need to place data values into equal-width or predefined 
+    bins. This function is useful for going from a continuous variable to a
```
Maybe restructure so doesn't repeat below: `..values into equal-width or predefined bins. The function supports both types through input arguments, This function is useful for...`

attack68 commented on pandas/core/reshape/tile.py:
```diff
@@ -294,11 +296,14 @@ def qcut(
-    on sample quantiles. For example 1000 values for 10 quantiles would
-    produce a Categorical object indicating quantile membership for each data point.
+    Transforms a variable with a large number of possibly continuous values 
+    into a variable with a smaller number of quantile-based intervals, or bins.
```
`Label values into equal-quantity or predefined quantile intervals, or bins.`

attack68 commented on pandas/core/reshape/tile.py:
```diff
@@ -294,11 +296,14 @@ def qcut(
+    into a variable with a smaller number of quantile-based intervals, or bins.
+    
+    Use `qcut` when you need to place values into equal (by 
+    the number of placed values) of unequal quantile-based bins. For example, 
```
`place values into equal-quantity or predefined quantile-based bins, all of likely unequal-width.`

attack68 commented on pandas/core/reshape/tile.py:
```diff
@@ -294,11 +296,14 @@ def qcut(
+    being assigned to the lowest bin, the next lowest 100 values being assigned 
+    to the next lowest bin, etc. 
 
     Parameters
```
Can you add a `see also` beneat returns.

```
See Also
---------
`pandas.cut` : Label values into pre-defined or equal-width intervals, or bins
```

attack68 commented on pandas/core/reshape/tile.py:
```diff
@@ -41,13 +41,15 @@ def cut(
+    categorical variable. For example, `cut` could convert a continuous age 
+    variable into age ranges variable. The function supports binning into a 
+    given number of equal-width intervals, or into pre-specified intervals.
+    The intervals can be provided either as intervals or as interval edges.
```
Can you add a see also beneath returns.

```
See Also
---------
`pandas.qcut` : Label values into equal-quantity or predefined quantile intervals, or bins.
```

attack68 reviewed (changes requested):
thanks for taking a look at this. couple of comments inline.

also if you run `black pandas` at the prompt should fixed some of the failing tests.

Ceay1 committed:
Update description of cut and qcut

Clarified/expanded description of cut and qcut.

Ceay1 committed:
Update description of cut and qcut functions

Update description of cut and qcut functions

Ceay1 commented:
I believe I have addressed all the suggestions. Please let me know if something else needs to be done on my end.

attack68 reviewed (changes requested):
one further comment.

also my comment about see also: `pandas.cut` and `pandas.qcut` might not be needed becuase modules are in same relative location so `cut` and `qcut` is fine I think generating links..

I dont have merge rights.. need a core developer to review and offer opinion also..

Ceay1 committed:
Update cut and qcut description

Update cut and qcut description

Ceay1 commented on pandas/core/reshape/tile.py:
```diff
@@ -294,11 +295,14 @@ def qcut(
-    on sample quantiles. For example 1000 values for 10 quantiles would
-    produce a Categorical object indicating quantile membership for each data point.
+    Use `qcut` when you need to place values into equal (by the number of placed
+    values) of unequal quantile-based bins. For example, placing 1000 values into
```
Done! Thank you for guiding me.

jreback commented on pandas/core/reshape/tile.py:
```diff
@@ -327,6 +331,13 @@ def qcut(
+    cut : Transform values into pre-defined or equal-width intervals (bins).
+    Categorical : Array type for storing data that come from a
+        fixed set of values.
+    Series : One-dimensional array with axis labels (including time series).
```
not sure how a reference to Series is relevant here (its true this can return a series but this is cluttering things)

Ceay1 replied:
Done, updated! Series - dropped, Wiki reference - added.

jreback commented:
>  A colleague of mine complained about the quality of the cut description. I took a chance to update the description of cut and qcut. If there are some suggestions to improve the description - please let me know.

would more examples help aside from the updated descriptions? 
can you add a link to wikipedia entries? https://en.wikipedia.org/wiki/Discretization_of_continuous_features or similar might be useful to point to.

Ceay1 committed:
Further update description of cut and qcut 

1. Added reference to Wiki (Discretization_of_continuous_features)
2. Dropped Series from See Also section fo qcut

Ceay1 committed:
Corrected Wikipedia link

Corrected Wikipedia link

jreback commented:
sounds good will look soon

Ceay1 commented:
> This pull request is stale because it has been open for thirty days with no activity. Please update or respond to this comment if you're still interested in working on this.

This pull request is limited in scope to updated descriptions of qcut and cut. All comments on the description of the cut and qcut have been addressed. Further comments (eg wrt additional examples) will be addressed in subsequent pull requests.

attack68 reviewed (approved):
lgtm

lithomas1 commented:
ping @jreback

jreback reviewed (changes requested):
this is on the 1.2.x branch. pls re-do to push to the master branch (or close and open a new one)

datapythonista commented:
@Ceay1 can you please open a new PR that  targets master, instead of the branch `1.2.x`, so we can merge? Closing this one.
